### PR TITLE
Add heap parameter to docs script for analytics

### PR DESCRIPTION
**Pull request by stichbury**
## Description

My attempt to set up Heap. Needs to be verified by someone who understands what they're doing.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: docs/source/_templates/layout.html [hunk 1]
```diff
@@ -32,6 +32,9 @@
 {%- block extrahead %}
 <script type="text/javascript">
   window.heap=window.heap||[],heap.load=function(e,t){window.heap.appid=e,window.heap.config=t=t||{};var r=document.createElement("script");r.type="text/javascript",r.async=!0,r.src="https://cdn.heapanalytics.com/js/heap-"+e+".js";var a=document.getElementsByTagName("script")[0];a.parentNode.insertBefore(r,a);for(var n=function(e){return function(){heap.push([e].concat(Array.prototype.slice.call(arguments,0)))}},p=["addEventProperties","addUserProperties","clearEventProperties","identify","resetIdentity","removeEventProperty","setEventProperties","track","unsetEventProperty"],o=0;o<p.length;o++)heap[p[o]]=n(p[o])};
+
+  heap.load({{ "522942930" }});
```
Comment (astrojuanlu) [inline docs/source/_templates/layout.html:36]: I think to mirror what's happening on Kedro, we'd need to do something like

https://github.com/kedro-org/kedro/blob/c3c93cb4786b4fd38c62baa2ec7f6fc73a63791e/docs/source/_templates/layout.html#L6

and then 

https://github.com/kedro-org/kedro/blob/c3c93cb4786b4fd38c62baa2ec7f6fc73a63791e/docs/source/conf.py#L488-L518

I can take a stab at this but not today

Comment (stichbury) [reply docs/source/_templates/layout.html:36]: Thanks! I don't believe we need `env_override` though, do we? I think it's used to differentiate `stable` and `latest` but for these two subprojects I don't see that as particularly desirable or important.

**Comment by stichbury** [comment]
@astrojuanlu @tynandebold Please can I check on the progress of this in my absence, if any?

**Comment by tynandebold** [comment]
No progress from me. I've just added it to our new sprint, so we're tracking it now and will get to it soon.

File: docs/source/_templates/layout.html (hunk 1 above)
Comment (tynandebold) [reply docs/source/_templates/layout.html:36]: I think it's used to differentiate between `development` and `production`, which is more important. I would argue we probably don't need that though. The number of us who'll be previewing the Viz docs and inflating the number of "real" users will be near zero.

**Review by tynandebold — APPROVED** [review]
This is working as it should be for me. I noted in a comment we're not switching between dev and prod environments, and I don't think we need to.

Here are the events coming through:

Screenshot 2023-11-01 at 15 34 06

**Pull request merged**